### PR TITLE
[GUI] Enable panning in dependency graph view

### PR DESCRIPTION
**Pull request by lvk88**
This PR makes it possible to drag the dependency graph using the
left mouse button.

See issue #3896.

Forum discussion:
https://forum.freecadweb.org/viewtopic.php?f=3&t=34791

Note that this PR only implements dragging by the left mouse button, and does not use the navigation mode of the 3D View set by the user. I had a look on some other "2D style" workbenches (Drawing, TechDraw, Image), and none of those seemed to use the 3D navigation modes. There seem to be no easy way to gather the navigation modifiers from those classes that inherit from `Gui::NavigationStyle`, because they seem to be implemented directly in `processEvent(...)`. I am also new to FreeCAD development, so I might be wrong. In this case, let me know and I will adjust accordingly.

- [x]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [x]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [x]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [x]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [x]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the [FreeCAD bug tracker issue number](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) in case a particular commit solves or is related to an existing issue on the tracker. Ex: `Draft: fix typos - fixes #0004805`
---

### CODE REVIEW

**Comment by luzpaz** [comment]
Proposed change:   

FreeCAD-panning-depgraph

**Comment by hyarion** [comment]
May I suggest changing "dragging" to "panning", both in code and the title of the pull request?

**Comment by lvk88** [comment]
> May I suggest changing "dragging" to "panning", both in code and the title of the pull request?

Sure, no problem

**Comment by sasobadovinac** [comment]
If the code change is quite safe, lets get this in to 0.19 :)

File: src/Gui/GraphvizView.cpp
Comment (lvk88) [reply src/Gui/GraphvizView.cpp]: You are right, it should be `Graphviz`, I changed it. Thanks for the hint!

File: src/Gui/GraphvizView.cpp
Comment (lvk88) [reply src/Gui/GraphvizView.cpp]: This is the matter of preference of course. I usually prefer if the member variables of a class are distinguished in some way, so that if I look at code, I immediately know from the name of the variable whether it was a parameter or belongs to the class.

But definitely, you are right, it is better to stick to the style of the surrounding parts, I removed the suffixes accordingly.

On a side note, I would try to avoid using `_` as a prefix, as it is reserved in the language.

**Comment by lvk88** [comment]
> If the code change is quite safe, lets get this in to 0.19 :)

That would be definitely cool! How would this work? Once this is merged, do I cherry pick my commits and open a new PR onto the 0.19 branch? Or would someone else do that?

File: src/Gui/GraphvizView.cpp
Comment (hyarion) [reply src/Gui/GraphvizView.cpp]: My personal preference is that code should be kept as uniform as possible but it's really difficult without a defined coding convention. So I'm trying to stick with the style with the scope, file, and/or module. It's not always obvious though :)

**Comment by hyarion** [comment]
The code looks good now 👍 
I haven't been able to test it yet as I manage to break my build environment and I'm still trying to piece it together..
So it would be nice if someone could give it a test and verify that it works as intended and approve if it does :)

> > If the code change is quite safe, lets get this in to 0.19 :)
> 
> That would be definitely cool! How would this work? Once this is merged, do I cherry pick my commits and open a new PR onto the 0.19 branch? Or would someone else do that?

The core devs will handle it when merging, if they think it should be included.

**Comment by lvk88** [comment]
Thanks for the quick and clear review! :)

Let's see if someone feels up for testing it.

**Review by hyarion — APPROVED** [review]
✅ Code looks good
✅ Works as described
✅ No one objects in the forum thread

I think this PR can be merged.

**Comment by wwmayer** [comment]
> That would be definitely cool! How would this work? Once this is merged, do I cherry pick my commits and open a new PR onto the 0.19 branch? Or would someone else do that?

Usually when making a release we don't add new features but only make bug fixes. But here we can make an exception as it's a very helpful improvement. I cherry-picked the change to the 0.19 release branch.

> My personal preference is that code should be kept as uniform as possible but it's really difficult without a defined coding convention. So I'm trying to stick with the style with the scope, file, and/or module. It's not always obvious though :)

I started a discussion last summer but unfortunately a lot of people were against fix coding conventions. So, at the moment one of the few restrictions is that for indentations we use four spaces while tabs are not recommended.
As coding convention you can apply that of the Qt project.

I squashed the tree commits into a single commit.

**Comment by luzpaz** [comment]
@wwmayer is there a way to add this commit to the 0.19 release without breaking all the links we have pointing to the stable from downstream and from our wiki and website? In more details, right now as I imagine, 0.19 stable builds will all be updated to include this commit and thereby changing filename of the download link to include either the revision number or the git hash. Can we use point releases instead that way the pre-existing stable link is maintained for 0.19.0 and still retrievable in addition to a new builds in 0.19.1 ?

**Comment by wwmayer** [comment]
Sorry, but I don't the answers to your questions :/

**Pull request closed without merging**
